pkg/controller/util/node/controller_utils.go

```diff
@@ -159,10 +159,11 @@ func MarkPodsNotReady(kubeClient clientset.Interface, pods []*v1.Pod, nodeName s
 // RecordNodeEvent records a event related to a node.
 func RecordNodeEvent(recorder record.EventRecorder, nodeName, nodeUID, eventtype, reason, event string) {
 	ref := &v1.ObjectReference{
-		Kind:      "Node",
-		Name:      nodeName,
-		UID:       types.UID(nodeUID),
-		Namespace: "",
+		APIVersion: "v1",
```

Comment:
I'm not an expert on the k8s api versioning... but just want to check there isn't a risk that the `APIVersion` will increase but we will forget to update this hard coded reference? Is there a constant we can refer to, or method call we can use which will guarantee this value stays up to date?

Reply:
I searched over the codebase and I see that often APIVersion is omitted.

Reply:
This is sorta related to the nodeRef not being a real node object in the Kubelet. I think this patch is ok, but it definitely should be refactored at some point.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L531-L533